packages/jest-fake-timers/src/legacyFakeTimers.ts

```diff
@@ -390,6 +399,7 @@ export default class FakeTimers<TimerRef> {
   }
 
   private _checkFakeTimers() {
+    // @ts-expect-error: condition always returns 'true'
```

Comment:
Perhaps it would make sense to copy `_checkFakeTimers` implementation from modern timers? Or to factor out the `_checkFakeTimers` from modern timers into a separate file and to `bind` it from there with both implementations? I could send separate PR.

https://github.com/facebook/jest/blob/54420eb51baecaba7dff293770f8cb2579825731/packages/jest-fake-timers/src/modernFakeTimers.ts#L138-L153